inference-engine/src/transformations/src/transformations/common_optimizations/simplify_shape_of_sub_graph.cpp

```diff
@@ -191,6 +191,99 @@ ngraph::pass::SimplifyGatherShapeOf::SimplifyGatherShapeOf() {
     this->register_matcher(m, callback);
 }
 
+NGRAPH_RTTI_DEFINITION(ngraph::pass::SimplifySecondInputOfReshape, "SimplifySecondInputOfReshape", 0);
+
+ngraph::pass::SimplifySecondInputOfReshape::SimplifySecondInputOfReshape() {
+    MATCHER_SCOPE(SimplifySecondInputOfReshape);
+    const auto input = ngraph::pattern::any_input();
+    const auto concat = ngraph::pattern::wrap_type<opset3::Concat>(pattern::consumers_count(1));
+    const auto reshape_pattern = ngraph::pattern::wrap_type<opset3::Reshape>({ input, concat });
```

Comment:
any_input -> ShapeOf -> Gather -> Concat -> Reshape
                 \-> Or(FQ) ---------------------------/

Please don't limit number of in-out nodes of Concat

Reply:
Check on consumers count of Concat was removed, thanks.
We can't match all pattern because current matcher doesn't support case when Gather is an arbitrary input of Concat. We have to add to the pattern only Reshape and  Concat and check the rest of the conditions in the callback.